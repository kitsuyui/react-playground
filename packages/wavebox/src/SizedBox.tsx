import type React from 'react'

/**
 * SizedBox component that wraps its children in a div with specified width and height.
 * @param props - The properties for the SizedBox component.
 * @param props.width - The width of the box.
 * @param props.height - The height of the box.
 * @returns {React.JSX.Element} The rendered SizedBox component.
 */
export const SizedBox = (props: {
  width: number
  height: number
  children: React.ReactNode
}): React.JSX.Element => {
  const { children, width, height } = props
  return (
    <div
      style={{
        width,
        height,
        overflow: 'hidden',
        position: 'relative',
      }}
    >
      {children}
    </div>
  )
}
