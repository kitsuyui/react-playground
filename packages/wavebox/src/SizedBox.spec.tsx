import { describe, expect, it } from 'vitest'

import { SizedBox } from './SizedBox'
import { render } from '@testing-library/react'

describe('SizedBox', () => {
  it('renders with specified width and height', () => {
    const { asFragment } = render(
      <SizedBox width={100} height={200}>
        <div>Test Content</div>
      </SizedBox>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
