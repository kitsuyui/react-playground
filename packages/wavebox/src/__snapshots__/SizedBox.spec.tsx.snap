// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SizedBox > renders with specified width and height 1`] = `
<DocumentFragment>
  <div
    style="width: 100px; height: 200px; overflow: hidden; position: relative;"
  >
    <div>
      Test Content
    </div>
  </div>
</DocumentFragment>
`;
